mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/IndexedRowMatrix.scala

```diff
@@ -76,6 +76,23 @@ class IndexedRowMatrix(
   }
 
   /**
+   * Converts this matrix to a
+   * [[org.apache.spark.mllib.linalg.distributed.CoordinateMatrix]].
+   */
+  def toCoordinateMatrix(): CoordinateMatrix = {
+    val entries = rows.flatMap { row =>
+      val rowIndex = row.index
+      row.vector match {
+        case SparseVector(size, indices, values) =>
+          Array.tabulate(indices.size)(x => MatrixEntry(rowIndex, indices(x), values(x)))
```

Comment:
`Array` -> `Iterator`. This is inside a flat map. We don't need to create an temp Array.

`x` -> `i`?